functions/src/modules/CR.functions.ts

```diff
@@ -0,0 +1,85 @@
+import * as functions from 'firebase-functions';
+import * as helpers from '../helpers';
+import GoogleSheets from '../services/GoogleSheets';
+import * as _ from "lodash";
+
+export const copySubmissionsToProcessing = functions
+  .pubsub
+  .topic('daily-tick')
+  .onPublish(async (message) => {
+    const onlineSheet = await
+      (new GoogleSheets(functions
+        .config()
+        .cr
+        .submissions
+        .spreadsheet
+        .id)
+      ).useSheet('Online');
+    const onlineSheetRows = await onlineSheet.getRows();
+
+    const destSpreadsheet = new GoogleSheets(functions.config().cr.processing.spreadsheet.id);
+    const destSheets = {};
+
+    const sheetNames = await destSpreadsheet.getSheetNames();
+
+    // used to hold `Submission subserials` of each sheet at the dest spreadsheet
+    const submissionSerials = {};
+
+    // Used to hold `Audio file names` of each sheet, to help look for the right place
+    // to insert a new submission
+    const audioFileNames = {};
+
+    // Holds the rows read from the `Online` sheet
+    // grouped by the sheet (list) name
+    const valuesToAppend = {};
+
+    const insertions = {};
+
+    for (const row of onlineSheetRows) {
+      const list = helpers.extractListFromFilename(row['Audio File Name']);
+
+      if (!destSheets[list]) {
+        // If the current sheet (list) doesn't exist on the target spreadsheet
+        if (sheetNames.indexOf(list) < 0)
+          console.warn(`Sheet ${list} of row ${row['Audio File Name']} does not exist at the destination spreadsheet!`);
+
+        destSheets[list] = await destSpreadsheet.useSheet(list);
+      }
+
+      if (!submissionSerials[list]) {
+        submissionSerials[list] = await destSheets[list].getColumn('Submission Serial');
+        audioFileNames[list] = await destSheets[list].getColumn('Audio File Name');
+      }
+
+      if (submissionSerials[list].indexOf(row['Submission Serial']) > -1)
+        continue;
+
+      // Find the right place to insert the row
+      const insIndex = _.sortedIndex(submissionSerials[list], row['Submission Serial']);
+
+      if (!insertions[list])
+        insertions[list] = [];
+
+      if (insertions[list][insIndex])
+        insertions[list][insIndex]++;
+      else
+        insertions[list][insIndex] = 1;
+
+      if (!valuesToAppend[list])
+        valuesToAppend[list] = [];
+
+      const insertionSum = insertions[list].reduce((acc, curr) => curr ? curr : 0, 0);
```

Comment:
I believe the logic here is incorrect. You need to sum all the insertions for `index <= originalIndex`. And let's call this variable `indexCorrection`.